src/cromshell/status/command.py

```diff
@@ -114,34 +104,48 @@ def main(config, workflow_id):
     return ret_val
 
 
-def get_metadata_status_summary(workflow_metadata):
-    """Get the status for each call in a workflow
-    and the frequency of those statuses"""
-    # workflow_metadata holds the workflow metadata as a dictionary
-    tmp_execution_status = []
-
-    # For each call in the metadata dictionary create a shortened summary containing
-    # the call name and status
-    for call in workflow_metadata["calls"]:
-        call_element = f'"{call}": '
-        execution_statuses = []
-
-        # For each call name add the number of execution_status to a list
-        for instanceDic in workflow_metadata["calls"][call]:
-            execution_statuses.append(instanceDic["executionStatus"])
-
-        # Create a key and value pair for the execution status and number of times
-        # it is seen
-        status_count = Counter(execution_statuses)
-        # For each status for the call add the call name and status name and count
-        # to list
-        for status in set(execution_statuses):
-            status_element = f'{{"{status}": {status_count[status]}}}'
-            tmp_execution_status.append(f"{call_element}{status_element}")
-
-    tmp_execution_status_json = ", ".join(tmp_execution_status)
-    tmp_execution_status_json = "{" + tmp_execution_status_json + "}"
-    return [json.loads(tmp_execution_status_json)]
+def check_for_failure(metadata: dict):
+    """Checks a workflow metadata dictionary for failing statuses
+    Returns True to indicate workflow or some task(s) has failed"""
+    workflow_failed = False
+    # If the given dictionary contains a 'status' key and has
+    # value of "Failed" then exit the function returning
+    # True to indicate workflow has failed
+    if "status" in metadata:
+        if metadata["status"] == "Failed":
+            workflow_failed = True
+            return workflow_failed
+    # If the dictionary does not contain a failed value for
+    # its status key or if status key does does not exist
+    # then we'll iterate through this loop in search for
+    # task status or subworkflows.
+    for key, value in metadata.items():
```

Comment:
It seems like you ignore the key, so why not just traverse the values?

Reply:
leftover stuff from previous code. changed